crates/wasmtime/src/runtime/types.rs

```diff
@@ -1241,6 +1294,20 @@ impl StorageType {
         self.as_val_type().unwrap()
     }
 
+    /// Unpack this (possibly packed) storage type into a full `ValType`.
+    ///
+    /// If this is a `StorageType::ValType`, then the inner `ValType` is
+    /// returned as-is.
+    ///
+    /// If this is a packed `StorageType::I8` or `StorageType::I16, then a
+    /// `ValType::I32` is returned.
+    pub fn unpack(&self) -> Cow<ValType> {
+        match self {
+            StorageType::I8 | StorageType::I16 => Cow::Owned(ValType::I32),
+            StorageType::ValType(ty) => Cow::Borrowed(ty),
+        }
+    }
```

Comment:
Could this return `&ValType` by using `const I32: ValType = ValType::I32;` and `&I32` for the i8/i16 case?

Reply:
Good call